### PR TITLE
[RFC] fix inproperly resetted tty issue under Windows

### LINKED ISSUE #10668: Windows: python shell has no interaction after quiting neovim
- `nvim --version`: NVIM v0.4.0-1487-g65578693b
- `vim -u DEFAULTS` (version: 8.1) behaves differently? yes
- Operating system/version: Windows 10 1903
- Terminal name/version: Alacritty 0.3.3
- `$TERM`: xterm-256color

### Steps to reproduce using `nvim -u NORC`

1. `nvim -u NORC`
2. exit neovim
3. open python shell
4. no input will be sent to python shell now after

Powershell works fine, once exited python, powershell continues to work.

### Actual behaviour
python shell has no input

### Expected behaviour
 python shell should work fine after quiting neovim

### PR DESCRIPTION
**Pull request by Escapingbug**
fixes #10668

### CODE REVIEW

**Comment by justinmk** [comment]
Seems like  the counterpart to 8072f085d2ed. For extra thoroughness @Escapingbug could you check if #10668 does _not_ happen if you build the commit before (parent of) 8072f085d2ed?

File: src/nvim/event/stream.c [hunk 1]
```diff
@@ -97,6 +97,10 @@ void stream_close(Stream *stream, stream_close_cb on_stream_close, void *data)
   stream->close_cb = on_stream_close;
   stream->close_cb_data = data;
 
+#ifdef WIN32
+  uv_tty_set_mode(&stream->uv.tty, UV_TTY_MODE_NORMAL);
```
Comment (justinmk) [inline src/nvim/event/stream.c]: `stream->uv` may not be a TTY stream, so we should check it somehow.

```suggestion
  if (UV_TTY == uv_guess_handle(stream->fd)) {
    uv_tty_set_mode(&stream->uv.tty, UV_TTY_MODE_NORMAL);
```

**Comment by Escapingbug** [comment]
> 
> 
> Seems like the counterpart to [8072f08](https://github.com/neovim/neovim/commit/8072f085d2edc254334af52261c6ff68d9143834). For extra thoroughness @Escapingbug could you check if #10668 does _not_ happen if you build the commit before (parent of) [8072f08](https://github.com/neovim/neovim/commit/8072f085d2edc254334af52261c6ff68d9143834)?

It seems true. And my last build shows that this truely will reintroduce that bug. I'm now stuck with strange building error, can't figure out how to deal with it for now..

File: src/nvim/event/stream.c (hunk 1 above)
Comment (Escapingbug) [reply src/nvim/event/stream.c]: It is now checked

**Comment by justinmk** [comment]
Merged, thank you @Escapingbug ! Glad you found this before the 0.4 release.

**Pull request closed without merging**